dojo/tools/sonarqube/parser.py

```diff
@@ -24,14 +23,26 @@ def get_label_for_scan_types(self, scan_type):
 
     def get_description_for_scan_types(self, scan_type):
         if scan_type == "SonarQube Scan":
-            return "Aggregates findings per cwe, title, description, file_path. SonarQube output file can be imported in HTML format or JSON format. Generate with https://github.com/soprasteria/sonar-report version >= 1.1.0, recommend version >= 3.1.2"
+            return "Aggregates findings per cwe, title, description, file_path. SonarQube output file can be imported in HTML format or JSON format. You can get the JSON output directly if you use the SonarQube API or generate with https://github.com/soprasteria/sonar-report version >= 1.1.0, recommend version >= 3.1.2"
         else:
             return "Import all findings from sonarqube html report or JSON format. SonarQube output file can be imported in HTML format or JSON format. Generate with https://github.com/soprasteria/sonar-report version >= 1.1.0, recommend version >= 3.1.2"
 
     def get_findings(self, filename, test):
-        if filename.name.strip().lower().endswith(".json"):
+        if filename.name.endswith(".json"):
             json_content = json.load(filename)
-            return self.get_json_items(json_content, test, self.mode)
+            if json_content.get("date") and json_content.get("projectName") and json_content.get("hotspotKeys"):
+                return SonarQubeSoprasteriaJSON().get_json_items(json_content, test, self.mode)
+            elif json_content.get("paging") and json_content.get("components"):
+                return SonarQubeRESTAPIJSON().get_json_items(json_content, test, self.mode)
+            else:
+                return []
+        if filename.name.endswith(".zip"):
+            if str(filename.__class__) == "<class '_io.TextIOWrapper'>":
+                input_zip = zipfile.ZipFile(filename.name, 'r')
+            else:
```

Comment:
```suggestion

```

I think this can be removed. While `zipfile.Zipfile` [can indeed take a "path to a file (a string) ... or a path-like object"](https://docs.python.org/3/library/zipfile.html#zipfile-objects), I don't know what that would mean in the context of a file passed in from an HTTP request to the API or UI, and I suspect it would not behave as intended here.

Reply:
No, it can't be removed. The if clause is if you upload the file in the GUI, and the else clause is for the unittest.